src/machines/__tests__/commerce-machine.test.ts

```diff
@@ -142,6 +203,27 @@ test('it can apply PPP coupon when available', async () => {
   expect(commerceService.state.context.priceId).toEqual(undefined)
 })
 
+test('it recognizes an applied default coupon', async () => {
+  const mockedCommerceMachine = commerceMachine.withConfig({
+    services: {
+      fetchPricingData: async (_context) => {
+        return Promise.resolve(pricingResponseWithDefaultApplied)
+      },
+    },
+  })
+
+  const commerceService = interpret(mockedCommerceMachine)
+
+  commerceService.start()
+
+  await sleep(0)
+
+  // expect(commerceService.state.context.priceId).toEqual(undefined)
+  expect(commerceService.state.context.couponToApply?.couponCode).toEqual(
+    'QJVIXHB6',
+  )
+})
```

Comment:
These tests still need to be refactored, but this works for now and ensures that the appropriate context gets set.